File: blinkreceipt-demo/BlinkReceiptAccountLinking/src/main/java/com/blinkreceipt/linking/BlinkApplication.java
package com.blinkreceipt.linking;

import android.app.Application;
import android.util.Log;

import androidx.annotation.NonNull;

import com.microblink.core.InitializeCallback;
import com.microblink.linking.BlinkReceiptLinkingSdk;

public class BlinkApplication extends Application {

    private final String TAG = "AccountLinking";

    @Override
    public void onCreate() {
        super.onCreate();

        BlinkReceiptLinkingSdk.initialize(this, new InitializeCallback() {

            @Override
            public void onComplete() {
                Log.d(TAG, "account linking initialization complete");
            }

            @Override
            public void onException(@NonNull Throwable throwable) {
                Log.e(TAG,  "failure in initialize", throwable);
            }

        });
    }
}

File: blinkreceipt-demo/BlinkReceiptEarnings/src/main/java/com/blinkreceipt/earnings/BlinkApplication.kt
package com.blinkreceipt.earnings

import android.app.Application
import android.util.Log
import com.microblink.core.InitializeCallback
import com.microblink.earnings.BlinkReceiptEarningSdk

class BlinkApplication : Application() {

    override fun onCreate() {
        super.onCreate()

        BlinkReceiptEarningSdk.initialize(this, object : InitializeCallback {

            override fun onComplete() {
                Log.d(TAG, "initialize complete" )
            }

            override fun onException(e: Throwable) {
                Log.e(TAG, "failure in initialize", e )
            }

        })
    }

    private companion object {
        const val TAG = "EarningsApplication"
    }
}